src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+            Requires.NotNull(activeConfiguredProject, nameof(activeConfiguredProject));
+            Requires.NotNull(lockService, nameof(lockService));
+
+            _vsProject = vsProject;
+            _activeConfiguredProject = activeConfiguredProject;
+            _lockService = lockService;
+            _threadingService = threadingService;
+
+            AddEventSource(this);
+        }
+
+        private ConfiguredProject ConfiguredProject => _activeConfiguredProject.Value;
+
+        public string Item(int lIndex)
+        {
+            return _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.ReadLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    var importsList = project.GetItems(importItemTypeName)
+                                             .Select(p => p.EvaluatedInclude)
+                                             .ToList();
+                    return importsList.ElementAt(lIndex - 1);
+                }
+            });
+        }
+
+        public void Add(string bstrImport)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    project.AddItem(importItemTypeName, bstrImport);
+                }
+            });
+        }
+
+        public void Remove(object index)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    var importProjectItem = project.GetItems(importItemTypeName)
+                                                   .ElementAt(((int)index - 1));
+                    project.RemoveItem(importProjectItem);
+                }
+            });
+        }
+
+        public IEnumerator GetEnumerator()
+        {
+            return _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.ReadLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    return project.GetItems(importItemTypeName)
+                                  .Select(p => p.EvaluatedInclude)
+                                  .GetEnumerator();
```

Comment:
No, we should make this work without that. You cannot expect consumers that use VSLangProj to take a read lock.